lib/transport_web/templates/layout/_header.html.eex

```diff
@@ -12,7 +12,7 @@
             <span><%= gettext "Sign Out" %></span>
           </a>
         <% else %>
-          <a class="navigation__link navigation__link--login" href="<%= page_path @conn, :login %>" role="link">
+          <a class="navigation__link navigation__link--login" href="<%= page_path @conn, :login, redirect_url: current_path(@conn) %>" role="link">
```

Comment:
Wouldn't it be easier to add a plug in the controller to assign the current path to the conn ?